src/components/ContentEditor/WidgetLiveEdit/WidgetLiveEdit.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { ContentSingularData } from '../../_data/ContentSingularData';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Button from '@material-ui/core/Button';
+import Chip from '@material-ui/core/Chip';
+import red from '@material-ui/core/colors/red';
+import green from '@material-ui/core/colors/green';
+import { WidgetTypes, ContentMapping } from '../../ContentMapping/ContentMapping';
+import { HistoryTypes } from '../../_debug/EditorHistory';
+import equal from 'deep-equal';
+import { EnvironmentContext } from '../../../contexts/EnvironmentContext/EnvironmentContext';
+import firebase from 'firebase';
+import { createMuiTheme } from '@material-ui/core';
+
+import './WidgetLiveEdit.css';
+
+type WidgetLiveEditProps = {
+  contentHash: string,
+  currYear: number,
+  pageToEdit: string,
+  user: firebase.User | null
+  editing: boolean
+};
+
+/**
+ * WidgetLiveEdit communicate with firebase and display whether a widget is
+ * currently being edited somewhere else.
+ */
+
+ export const WidgetLiveEdit: React.FC<WidgetLiveEditProps> = ({
+   contentHash, currYear, pageToEdit, user, editing
+ }) => {
+  let [message, setMessage] = useState<string>("safe to edit");
+
+  let widgetRef: firebase.database.Reference = firebase.database().ref(`${currYear}/LiveEditHistory/${pageToEdit}/${contentHash}`);
+
+  const updateOnce = () => {
+    widgetRef.update({
+      timestamp: firebase.database.ServerValue.TIMESTAMP,
+      editor: (user && user.email) || "Unknown user",
+      saved: false
+    });
+  };
+
+  const readTimestampOnce = () => {
+    widgetRef.once('value', function(snapshot) {
+      if (snapshot.val()) {
+        let diff: number = (Date.now() - snapshot.val().timestamp) / 1000;
+        let saved: boolean = snapshot.val().saved;
+
+        if (!saved && diff < 60) {
```

Comment:
where does `60` come from? It might be better to split it into its own constant so we know what it is.